inc/ThirdParty/Hostings/OneCom.php

```diff
@@ -0,0 +1,269 @@
+		$this->options     = $options;
+	}
+
+	/**
+	 * Array of events this subscriber wants to listen to.
+	 *
+	 * @since 3.6.3
+	 *
+	 * @return array
+	 */
+	public static function get_subscribed_events() {
+		return [
+			'init'                                    => [
+				[ 'maybe_enable_cdn_option' ],
+				[ 'maybe_disable_cdn_option' ],
+			],
+			'rocket_cdn_reject_files'                 => 'exclude_from_cdn',
+			'rocket_disable_cdn_option_change'        => 'is_oc_cdn_enabled',
+			'rocket_cdn_settings_fields'              => 'disable_cdn_change',
+			'do_rocket_varnish_http_purge'            => 'is_varnish_active',
+			'rocket_varnish_field_settings'           => 'maybe_set_varnish_addon_title',
+			'rocket_display_input_varnish_auto_purge' => 'should_display_varnish_auto_purge_input',
+		];
+	}
+
+	/**
+	 * Check if one.com cdn is enabled.
+	 *
+	 * @return boolean
+	 */
+	public function is_oc_cdn_enabled(): bool {
+		return rest_sanitize_boolean( get_option( 'oc_cdn_enabled' ) );
+	}
+
+	/**
+	 * Enable CDN option.
+	 *
+	 * @return void
+	 */
+	public function maybe_enable_cdn_option() {
+
+		if ( ! $this->is_oc_cdn_enabled() ) {
+			return;
+		}
+
+		$this->get_cdn_options();
+
+		$cdn_url = $this->build_cname();
+
+		if ( $this->cdn ) {
+			return;
+		}
+
+		$this->cdn_cnames = [ $cdn_url ];
+		$this->cdn_zones  = [ 'all' ];
+
+		$this->update_cdn_options( 1, $this->cdn_cnames, $this->cdn_zones );
+
+		rocket_clean_domain();
+	}
+
+	/**
+	 * Disable cdn option.
+	 *
+	 * @return void
+	 */
+	public function maybe_disable_cdn_option() {
+
+		if ( $this->is_oc_cdn_enabled() ) {
+			return;
+		}
+
+		$this->get_cdn_options();
+
+		if ( ! $this->cdn ) {
+			return;
+		}
+
+		$cdn_url = $this->build_cname();
+
+		// Remove CDN CNAME.
+		$cname_index_key = array_search( $cdn_url, $this->cdn_cnames, true );
+		unset( $this->cdn_cnames[ $cname_index_key ] );
+
+		// Remove CDN Zone.
+		unset( $this->cdn_zones[ $cname_index_key ] );
+
+		$this->update_cdn_options( 0, $this->cdn_cnames, $this->cdn_zones );
+
+		rocket_clean_domain();
+	}
+
+	/**
+	 * Exclude files from being rewritten.
+	 *
+	 * @param array $files Array of files to be excluded.
+	 * @return array
+	 */
+	public function exclude_from_cdn( array $files ): array {
+
+		if ( ! $this->is_oc_cdn_enabled() ) {
+			return $files;
+		}
+
+		$files[] = '/wp-includes/(.*)';
+
+		return $files;
+	}
+
+	/**
+	 * Disable CDN option change.
+	 *
+	 * @param array $settings CDN field settings data.
+	 * @return array
+	 */
+	public function disable_cdn_change( array $settings ): array {
+		if ( ! $this->is_oc_cdn_enabled() ) {
+			return $settings;
+		}
+
+		$settings['cdn']['container_class'][]      = 'wpr-isDisabled';
+		$settings['cdn']['input_attr']['disabled'] = 1;
+
+		return $settings;
+	}
+
+	/**
+	 * Purge varnish cache.
+	 *
+	 * @return boolean
+	 */
+	public function should_display_varnish_auto_purge_input(): bool {
+		return $this->is_varnish_active() ? false : true;
```

Comment:
```suggestion
		return $this->is_varnish_active();
```

The value returned by the method is already a boolean, you can use it directly

Reply:
To catch false positive when one.com varnish is active (`is_varnish_active()` returns true) we want to return false so as to deactivate the toggle switch for our varnish addon.

Reply:
You can reverse the condition with `!` in front. I updated the suggestion to reflect that